model/src/main/kotlin/utils/FileArchiverStorage.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package org.ossreviewtoolkit.model.utils
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.Provenance
+
+/**
+ * A storage for file archives.
+ */
+interface FileArchiverStorage {
+    /**
+     * Return whether an archive corresponding to [provenance] exists.
+     */
+    fun hasArchive(provenance: Provenance): Boolean
+
+    /**
+     * Add the [archive][zipFile] corresponding to [provenance]. Overwrites any existing archive corresponding to
+     * [provenance].
+     */
+    fun addArchive(provenance: Provenance, zipFile: File)
```

Comment:
Just a thought (not sure how feasible this is): Would it make sense to base this interface on streams rather than files? This could make the creation of temporary zip files unnecessary in some cases. For instance, if Postgres is used as storage, streams can probably be used directly to read from and write to Blob objects. The streams can be wrapped by Zip streams to handle compression transparently.

Reply:
I considered doing that. The reasons I decided for files because (1) the handling seemed a bit simpler to me (2) performance wise it should not matter as the network is likely the bottleneck as opposed to the file creation (3) it seemed to be simpler to implemented bulk fetching (future PR) as that bulk fetch function could then just return `Map<Provenance, File?>`. E.g. returning hundreds of open streams seems not right so a implementation would need to be a bit smarter - that would probably be nice but also more effort to implement and maintain. All in all I'd prefer keeping it as is for now.